src/spikeinterface/core/generate.py

```diff
@@ -195,39 +246,53 @@ def synthesize_random_firings(
         Concatenated and sorted label vector
 
     """
-    if seed is not None:
-        np.random.seed(seed)
-        seeds = np.random.RandomState(seed=seed).randint(0, 2147483647, num_units)
-    else:
-        seeds = np.random.randint(0, 2147483647, num_units)
 
-    if isinstance(firing_rates, (int, float)):
-        firing_rates = np.array([firing_rates] * num_units)
+    rng = np.random.default_rng(seed=seed)
 
-    refractory_sample = int(refractory_period_ms / 1000.0 * sampling_frequency)
-    refr = 4
+    # unit_seeds = [rng.integers(0, 2 ** 63) for i in range(num_units)]
 
-    N = np.int64(duration * sampling_frequency)
+    # if seed is not None:
+    #     np.random.seed(seed)
+    #     seeds = np.random.RandomState(seed=seed).randint(0, 2147483647, num_units)
+    # else:
+    #     seeds = np.random.randint(0, 2147483647, num_units)
 
-    # events/sec * sec/timepoint * N
-    populations = np.ceil(firing_rates / sampling_frequency * N).astype("int")
-    times = []
-    labels = []
-    for unit_id in range(num_units):
-        times0 = np.random.rand(populations[unit_id]) * (N - 1) + 1
+    if np.isscalar(firing_rates):
+        firing_rates = np.full(num_units, firing_rates, dtype="float64")
 
-        ## make an interesting autocorrelogram shape
-        times0 = np.hstack(
-            (times0, times0 + rand_distr2(refractory_sample, refractory_sample * 20, times0.size, seeds[unit_id]))
-        )
-        times0 = times0[np.random.RandomState(seed=seeds[unit_id]).choice(times0.size, int(times0.size / 2))]
-        times0 = times0[(0 <= times0) & (times0 < N)]
+    refractory_sample = int(refractory_period_ms / 1000.0 * sampling_frequency)
 
-        times0 = clean_refractory_period(times0, refractory_sample)
-        labels0 = np.ones(times0.size, dtype="int64") * unit_id
+    segment_size = int(sampling_frequency * duration)
 
-        times.append(times0.astype("int64"))
-        labels.append(labels0)
+    times = []
+    labels = []
+    for unit_ind in range(num_units):
+        n_spikes = int(firing_rates[unit_ind] * duration)
+        # we take a bit more spikes and then remove if too much of then
+        n = int(n_spikes + 10 * np.sqrt(n_spikes))
+        spike_times = rng.integers(0, segment_size, n)
+        spike_times = np.sort(spike_times)
+
+        if add_shift_shuffle:
+            ## make an interesting autocorrelogram shape
+            # this replace the previous rand_distr2()
+            some = rng.choice(spike_times.size, spike_times.size // 2, replace=False)
+            x = rng.random(some.size)
+            a = refractory_sample
+            b = refractory_sample * 20
+            shift = a + (b - a) * x**2
+            spike_times[some] += shift
+            times0 = times0[(0 <= times0) & (times0 < N)]
```

Comment:
The N here is not defined.

Reply:
Thanks @h-mayorquin 

@samuelgarcia can you make a PR to fix it?

Reply:
oups. thank for this check. I will fix this